### PR TITLE
Show 24h change for btc and fiat

### LINKED ISSUE #82: Feature Request: 24 hr change Settings
It would be nice to have the option to choose whether the 24 hr change data is based off of the BTC price or the USD price, or have two separate indicators for both.

### PR DESCRIPTION
**Pull request by Nasicus**
My first attempt is to simply show the 24h change for both currencies right next to the price - what do you guys think? I also think the `24h` label is enough to understand what the meaning behind the numbers are

screenshot_20180311-153638

edit: hmm the change fro BTC to almost all fiat currencies appears to be almost identical. Thought there might be an error in the code but it's probably correct.

@Levalicious what do you think?

Resolves #82

### CODE REVIEW

**Comment by Levalicious** [comment]
The information is definitely still communicated with just the 24h. As for the change, can't really see how they'd be the same... In the picture you're showing above, the CHF price change should be positive.

**Comment by Nasicus** [comment]
Well I find it also strange, but then it's crypto compare which returns something wrong, this is the call we make:

https://min-api.cryptocompare.com/data/pricemultifull?fsyms=ARK&tsyms=BTC,USD,EUR,GBP,KRW,CNY,JPY,AUD,CAD,RUB,INR,BRL,CHF,HKD,IDR,MXN

**Comment by Levalicious** [comment]
I just loaded that page, and it definitely has different values for RAW.ARK.USD.CHANGEPCT24HOUR and RAW.ARK.BTC.CHANGEPCT24HOUR so I'm not sure what would be causing the issue

**Comment by Nasicus** [comment]
@Levalicious  it has the same value for `CHANGEPCT24HOUR` .... maybe we need to calculate this change ourselves with the data we get...hmm

**Comment by Levalicious** [comment]
Looking a bit more, the calls for those same values from display instead of raw show the percentages you have, so I'm not sure why those would be in disagreement

**Comment by Nasicus** [comment]
So I tried to analyze this once more.
Let's take `USD` as our example.

The call looks like this:
https://min-api.cryptocompare.com/data/pricemultifull?fsyms=ARK&tsyms=USD
and returns something like this (omitted irrelevant stuff):

```
{"DISPLAY":{"ARK":{"USD":{"FROMSYMBOL":"ARK","TOSYMBOL":"$","MARKET":"CryptoCompare Index","PRICE":"$ 3.23","CHANGEPCT24HOUR":"3.28"}}}}
```
If we do the call for any other fiat currency we will always have the same `CHANGEPCT24HOUR` value.
As I quoted above already, in the [docs](https://www.cryptocompare.com/api/#-api-data-price-) is written the following 

> ... If the crypto does not trade directly into the toSymbol requested, BTC will be used for conversion. ...

So let's make the same call and prevent the auto conversion:
https://min-api.cryptocompare.com/data/pricemultifull?fsyms=ARK&tsyms=USD&tryConversion=false
```
{"Response":"Error","Message":"CCCAGG market does not exist for this coin pair (ARK-USD)","Type":1,"Aggregated":false,"Data":[]}
```

Ok. So that means, that most likely `BTC` is indeed used for the conversion.

What are the implications of this?
The price which we get in USD is probably correct, since (ARK => BTC => USD) is easy to calculate.
However the `CHANGEPCT24HOUR` is probably not easy or even possible to get via the `BTC` conversion. If they want to do that, they'd need to keep track of the previous days (for ARK => BTC => USD).

So long story short: I think the `CHANGEPCT24HOUR`  will NEVER be correct for any fiat currency (because no fiat currencies are directly supported on this API call).

Can someone confirm my thoughts? Or do I have something wrong?

Possible solutions:

- only show 24h change for BTC
- for certain currencies (USD for sure and maybe others), we could change the call so it takes another  exchange (parameter `e`) for the comparison. However this would result in multiple calls and we would need to know exactly which currencies support a direct conversion.

**Comment by Levalicious** [comment]
Is there an issue with calling RAW instead of display?

**Comment by Nasicus** [comment]
raw has the same data

**Comment by Levalicious** [comment]
Yesterday, it had different data.

**Comment by Nasicus** [comment]
well raw is just not rounded I think... for example:  `"CHANGEPCT24HOUR":3.2819954532355684` vs `"CHANGEPCT24HOUR":"3.28"`.... pretty sure it's the same, maybe you just oversaw it

**Comment by alexbarnsley** [comment]
@Nasicus I can see the same problems you're describing. BTC % (rounded) vs USD % (not rounded). I too am unsure how to resolve this, without calling a different API. If there's no resolution, maybe it's worth keeping it how it is 🤔

**Comment by Nasicus** [comment]
Well the problem is, that it's wrong at the moment. The shown % is always for BTC and that's not obvious or is it?

**Comment by alexbarnsley** [comment]
i can see what you're saying @Nasicus. would an alternative be to just rename the heading to "BTC 24h change"?

**Comment by Nasicus** [comment]
Yes this would be a suitable solution for now I think.

**Comment by alexbarnsley** [comment]
Okay, great. Do you want to possibly do a new PR for that? Make it a bit cleaner? :smile:

**Comment by Nasicus** [comment]
Yes will do that this evening 👍 

I'll close this one, but then we know what the current situation is and what problems we have.

**Comment by alexbarnsley** [comment]
Sounds great! Can link to this PR from the new one too. Thanks!

**Comment by Nasicus** [comment]
This is now partially fixed / addressed in #130 .
